documentation/AuthServer.md

```diff
@@ -25,56 +53,110 @@ The document types used by the authorization server include:
 
-#### Client
+#### GPII Tokens
+
+| Option | Type | Description | Default |
+| ------ | ---- | ----------- | ------- |
+| `id` | String | The GPII token id. Can be a UUID or any unique string. This value is saved into `_id` field in CouchDB/PouchDB. | None |
+| `type` | String | The document type for storing GPII tokens. | The value must be set to "gpiiToken". |
+| `gpiiToken` | String | The GPII token. | None |
+| `userId` | String | The user id that this GPII token belongs to. | None |
+
+#### GPII App Installation Clients
 
 | Option | Type | Description | Default |
 | ------ | ---- | ----------- | ------- |
 | `id` | String | The client id. Can be a UUID or any unique string. This value is saved into `_id` field in CouchDB/PouchDB. | None |
-| `type` | String | The document type for storing client information. | The value must be set to "client". |
+| `type` | String | The document type for storing the client type information. | The value must be set to "gpiiAppInstallationClient". |
 | `name` | String | The client name. | None |
 | `oauth2ClientId` | String | The unique identifier issued to a registered OAuth2 client by the authorization server. | None |
 | `oauth2ClientSecret` | String | Confidential shared secret between the client and the authorization server, used to verify the identity of the client. | None |
-| `redirectUri` | String | The URL on client's site where users will be sent after authorization. | None |
+| `userId` | Boolean | The id of the user who creates this client.| None |
 
-#### GPII Tokens
+#### GPII App Installation Authorizations
 
 | Option | Type | Description | Default |
 | ------ | ---- | ----------- | ------- |
-| `id` | String | The GPII token id. Can be a UUID or any unique string. This value is saved into `_id` field in CouchDB/PouchDB. | None |
-| `type` | String | The document type for storing GPII tokens. | The value must be set to "gpiiToken". |
-| `gpiiToken` | String | The GPII token. | None |
-| `userId` | String | The user id that this GPII token belongs to. | None |
+| `id` | String | The GPII app installation authorizations id. Can be a UUID or any unique string. This value is saved into `_id` field in CouchDB/PouchDB. | None |
+| `type` | String | The document type for storing GPII app installation authorizationss. | The value must be set to "gpiiAppInstallationAuthorization". |
+| `clientId` | String | The client id that this token is assigned to. | None |
+| `gpiiToken` | String | The GPII token that this token record is associated with. | None |
+| `accessToken` | String | The access token used to retrieved the protected user preferences. | None |
+| `expiresIn` | Number | The number of seconds left before the access token becomes invalid. The default value is 3600. | 3600 |
+| `revoked` | Boolean | Whether this token has been revoked. | false |
+| `expired` | Boolean | Whether this token has expired, meaning `timestampCreated` + `expiresIn` < now(). | false |

```

Comment:
It looks to me like view results are cached in CouchDB rather than calculated each time (though I'm not super familiar with CouchDB yet):

"If you have a lot of documents, that takes quite a bit of time and you might wonder if it is not horribly inefficient to do this. Yes, it would be, but CouchDB is designed to avoid any extra costs: it only runs through all documents once, when you first query your view. If a document is changed, the map function is only run once, to recompute the keys and values for that single document."

"The view result is stored in a B-tree, just like the structure that is responsible for holding your documents."

http://guide.couchdb.org/draft/views.html

By calculating the expired state ourselves outside of the database and managing the caching of it (by storing the `expired` boolean on the document) we run the risk of inconsistent data. I think we'd be better to calculate it in CouchDB and let CouchDB manage the caching.

I had a look at the `gpii.oauth2.isExpired` function. Isn't it just doing `timestampCreated + expiresIn < now()` but with some format and units conversion?